exercises/rest-api/canonical-data.json

```diff
@@ -0,0 +1,292 @@
+{
+    "exercise": "rest-api",
+    "version": "1.0.0",
+    "comments": [
+      "The state of the API database before the request is represented",
+      "by the input->database object. Your track may determine how this",
+      "initial state is set",
+      "The inputs and outputs are represented as arrays of strings to",
+      "improve readability in this JSON file.",
+      "Your track may choose whether to present the input as a single",
+      "string (concatenating all the lines) or as the list.",
+      "In most cases, it seems to make sense to expect the output as",
+      "a single string.",
+      "Additionally, your track may choose whether to enforce or ignore",
+      "whitespace in JSON strings."
+    ],
+    "cases": [
+        {
+            "description": "user management",
+            "cases": [
+                {
+                    "description": "no users",
+                    "property": "request",
+                    "input": {
+                        "database": {
+                            "users": []
+                        },
+                        "request_body": ["GET /users"]
+                    },
+                    "expected": [
+                        "{",
+                        "\"users\": []",
```

Comment:
Maybe for readability indent the contents in this string? So that would make it:

```json
"{"
"  \"users\": []",
"}"

Reply:
Seems reasonable.